packages/core/src/PartialExecutionUtils/__tests__/cleanRunData.test.ts

```diff
@@ -0,0 +1,86 @@
+import type { IRunData } from 'n8n-workflow';
+import { cleanRunData } from '../cleanRunData';
+import { DirectedGraph } from '../DirectedGraph';
+import { createNodeData, toITaskData } from './helpers';
+
+describe('cleanRunData', () => {
+	// ┌─────┐    ┌─────┐   ┌─────┐
+	// │node1├───►│node2├──►│node3│
+	// └─────┘    └─────┘   └─────┘
+	test('deletes all run data of all children and the node being passed in', () => {
+		// ARRANGE
+		const node1 = createNodeData({ name: 'Node1' });
+		const node2 = createNodeData({ name: 'Node2' });
+		const node3 = createNodeData({ name: 'Node3' });
+		const graph = new DirectedGraph()
+			.addNodes(node1, node2, node3)
+			.addConnections({ from: node1, to: node2 }, { from: node2, to: node3 });
+		const runData: IRunData = {
+			[node1.name]: [toITaskData([{ data: { value: 1 } }])],
+			[node2.name]: [toITaskData([{ data: { value: 1 } }])],
+			[node3.name]: [toITaskData([{ data: { value: 1 } }])],
```

Comment:
Should we perhaps use different values to distinguish from each others? Same for the other test cases

Reply:
Love that idea! Done